pages/gall/[id]/index.tsx

```diff
@@ -38,6 +59,10 @@ const hostAsLink = (len: number) => (h: GallHost, idx: number) => {
 const Gall = ({ species, imagePaths }: Props): JSX.Element => {
     const source = species ? species.speciessource.find((s) => s.useasdefault !== 0) : undefined;
     const [selectedSource, setSelectedSource] = useState(source);
+
+    // Initially sort the list of sources by most recent year.
+    species.speciessource.sort((a, b) => -a.source.pubyear.localeCompare(b.source.pubyear));
```

Comment:
Move this line after line 72 so that it will not get called if we are in a loading state.

Reply:
On my machine this change results in an eslint error complaining that "React Hooks must be called in the exact same order in every component render," claiming that it was called conditionally.
That being said, `yarn build` succeeds and it seems to work fine when locally hosted.
Is there an idiomatic way to handle this possible failure besides just loading the resource conditionally?

Reply:
You have to move the useState call above the conditional as well. I forgot to mention that. One of the rules of React hooks is that you can not define state (uses a hook) after a call that returns early.